bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -198,10 +200,19 @@ public void stateUpdated(Item item, State state) {
                 event.valuecolor = itemUIRegistry.getValueColor(w);
                 event.widgetId = itemUIRegistry.getWidgetId(w);
                 event.visibility = itemUIRegistry.getVisiblity(w);
+                // event.item contains data from the item including its state (in event.item.state)
                 event.item = EnrichedItemDTOMapper.map(item, false, null, null);
 
-                // adjust the state according to the widget type
-                event.item.state = itemUIRegistry.getState(w).toFullString();
+                // event.state is an adjustment of the item state to the widget type.
+                // In case this state is big and identical to the item state, an indirection to the
+                // item state is provided by setting its value to "#item.state#" in order to avoid
+                // duplicating in the event message a big amount of data.
+                String state = itemUIRegistry.getState(w).toFullString();
+                if ((state.length() > REDIRECT_TO_ITEM_STATE.length()) && state.equals(event.item.state)) {
```

Comment:
This looks dodgy. Is this the coded translation of "In case this state is big and identical to the item state"?
If so, why is anything bigger than 12 already "big"?

Reply:
Because @maggu2810 suggested this change yesterday ;) Before that, it was set to 128 bytes.
But considering your other suggestion, we don't care about that now.

Reply:
Yes, it has been suggested by me here: https://github.com/eclipse/smarthome/pull/4810#discussion_r160153919

WRT to @kaikreuzer comments above, not using a magic string constant at all would be better -- I agree.
But if we use a magic string constant, I still think it should be used as soon as it makes some difference and not only by a magic "128". :wink: 

So, if we can get rid off the magic string it would be better at all.

Reply:
Ah, thanks, I had missed that comment.

But ok, it seems we all agree with abandoning the REDIRECT_TO_ITEM_STATE string completely.
@lolodomo Once you have done this change, I think we should be good to merge.

Reply:
Ok.
I cannot do that before Friday in the best case.

Reply:
Ok, no problem - I would then schedule a first ESH stable build since the OH 2.2 release for next Monday, so that this change is included.